src/context/AppProvider/AppContext.tsx

```diff
@@ -145,7 +150,7 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
         }),
       )
     })()
-  }, [dispatch, wallet, supportedChains, isSnapInstalled])
+  }, [dispatch, wallet, supportedChains, isSnapInstalled, accountIdsByChainId])
```

Comment:
still feels like your first intuition may have been correct here. I don't think this will break anything, but certainly add extra overhead. rationale would be `isSnapInstalled` is already in the dependency array for metamask snaps and the only other new accounts we get are either via `AddAccountModal` or `LedgerChains` component, both of which already call `getAccount`, and upserts.

Whether or not we would want to rely on app context for the `getAccount` and `upsertPortfolio` calls and remove that notion from the modal/components would be another question, but as it stands, it doesn't seem like we need to react on accountIdsByChainId at this time.

Reply:
Yeah the idea was to not be too re-rerendery here as this is inherently risky in terms of overfetching. And realistically the non-reactive version wouldn't be an issue - if it was, the previous version wouldn't work as support would be wrongly detected as false as AccountIds are being populated. 

From the testing I've done, this guy is ran 2/3 times as AccountIds are populating which now means a no-op thw 1/2 additional runs fetching-wise. With that being said, I don't think `{forceRefetch: true}` was ever a good idea for this effect:

- if this RTK query didn't run yet, it will and fetch fresh data
- if it did run (e.g in a previous run), this will serve cached data

Happy to either go back to doing the non-reactive dance here or keep this, but either way we should probably keep `{forceRefetch: false}` IMO!

re: `Whether or not we would want to rely on app context for the getAccount and upsertPortfolio calls and remove that notion from the modal/components would be another question`

I'd say both firing `getAccount` RTK query is fine, as `<AppContext />` *should* be the main initiator regardless, with `<LedgerChains />` and `<AddAccountModal />` being explicit with `{forceRefetch: true}` in comparison to `<AppContext />`'s (now) false force refetching.

Reply:
Yeah those are fair and valid points that I can get behind. I just get a little hesitant with AppContext changes obviously. What do you think about an isolated PR for this change in behavior as a follow up just seeing as there are a lot of other things to worry about and focus on in this PR for now?

Reply:
Yeah after giving this more thought it's probably not worth the risk either way - as you've correctly pointed out, we do not *actually* need this, and excessive reactivity in `<AppContext />` is inherently dangerous - 9f1754266e